File: pset6/credit/credit.py
from cs50 import get_string 
import math

number = get_string("Number: ")
length = len(number)


SumOdd = 0
SumEven = 0

s = []
for i in range(length):
    s.append(int(number[i]))


if length % 2 == 1:
    for i in range(0, length, 2):
        SumOdd += s[i]
    for j in range(1, length, 2):
        if 2 * s[j] >= 10:
            t = math.floor((2 * s[j]) / 10) + (2 * s[j]) % 10
        else:
            t = 2 * s[j]
        SumEven += t
        
else:
    for i in range(1, length, 2):
        SumOdd += s[i]
    for j in range(0, length, 2):
        if 2 * s[j] >= 10:
            t = math.floor((2 * s[j]) / 10) + (2 * s[j]) % 10
        else:
            t = 2 * s[j]
        SumEven += t
        
print(SumOdd)
print(SumEven)

if (SumOdd + SumEven) % 10 != 0:
    print("INVALID")
elif s[0] == 4 and (length == 13 or length == 16):
    print("VISA")
elif s[0] == 3 and (s[1] == 4 or s[1] == 7) and length == 15: 
    print("AMEX")
elif s[0] == 5 and s[1] > 0 and s[1] < 6 and length == 16:
    print("MASTERCARD")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

{
    long c = get_long("Number:"); //get the credit card number
    int d = floor(c / 1000000000000);  //for VISA 13-digit
    int e = floor(c / 10000000000000); //for AMEX 15-digit
    int f = floor(c / 100000000000000); //for master 16-digit
    int g = floor(c / 1000000000000000); //for VISA 16-digit
    
    long a = c;
    long b = c;
    int i = 0;
    int j = 0;
    
    while (a > 1)
    {
        i = i + a % 10; 
        a = (a - a % 10 - 10 * (a / 10 % 10)) / 100; 
    } 
     
    while (b > 10)
    {
        j = j + (2 * (b / 10 % 10) % 10) + ((2 * (b / 10 % 10) / 10) % 10) ;     //need to think how to get the product digit for number more than 10
        b = (b - b % 10 - 10 * (b / 10 % 10)) / 100;  
        
    }

    
    
    if ((i + j) % 10 == 0)
    {
        if (e == 34 || e == 37)
        {
            printf("AMEX\n");
        }
    
        else if (!(f < 50) && !(f > 55))
        {
            printf("MASTERCARD\n");
        }
    
        else if (d == 4)
        {
            printf("VISA\n");
        }
    
        else if (g == 4)
        {
            printf("VISA\n");
        }
        else 
        {
            printf("INVALID\n");
        }
    }
   
    else 
    {
        printf("INVALID\n");
    }
}



//count if the number have 16 digits, otherwise invalid
//calculate if the nuber is valid
//determine if it is VISA, Master or AMEX, can use floor(c/10^15)
//print the result